types/react-google-recaptcha/index.d.ts

```diff
@@ -15,22 +16,32 @@ export { ReCAPTCHA };
 
 declare class ReCAPTCHA extends React.Component<ReCAPTCHAProps> {
     /**
-     * Resets the reCAPTCHA widget
+     * Resets the reCAPTCHA widget.
      */
     reset(): void;
+
     /**
-     * Programatically invoke the reCAPTCHA check. Used if the invisible reCAPTCHA is on a div instead of a button
+     * Programatically invoke the reCAPTCHA check. 
+     * Used if the invisible reCAPTCHA is on a div instead of a button.
      */
     execute(): void;
 
+    /**
+     * Programmatically invoke the challenge and return a promise that resolves
+     * to the token or errors (if encountered). Alternative approach to 
+     * execute() in combination with the onChange() prop.
+     * @return token | null | undefined
+     */
+    executeAsync(): Promise<string | null | undefined>
```

Comment:
@mocantimoteidavid Are you sure that both null and undefined can be returned? because according to the typings of onChange it's actually just string|null

Reply:
@RohrerF Actually it's because I wasn't too sure I've made it both null & undefined. 
I've searched for a more definite answer online, but wasn't able to find much. So just to be sure all cases would be covered by anyone using this, I've made it undefined too.

If you think `string | null` is enough, I'm happy to adjust it.

Reply:
the return of ```execute()``` and ```executeAsync()``` should be more or less the same so maybe the definition of ```execute()``` is incorrect since there is a [Path where it doesn't return anything](https://github.com/dozoisch/react-google-recaptcha/blob/7676c9b1bb268be74fba22d6f5d3760f6001a4b1/src/recaptcha.js#L27)

Reply:
To be honest I don't understand all the bits & pieces of how the component is behaving, but what it seems to me is that the `execute()` method actually doesn't return anything, so it's always 'void'. Also I don't think the `grecaptcha.execute(this._widgetId);` that it's calling returns anything.

So given this + the fact (because I tried) that `executeAsync()` does return the token string, I would argue that they are not returning the same thing.

Of course we shouldn't judge this by opinions, so because I have too many of these & not too many facts, I'm happy to change whatever we conclude it's correct.

Reply:
so we need the string|null for [reCAPTCHA v3](https://developers.google.com/recaptcha/docs/v3#programmatically_invoke_the_challenge) in this version it looks like a string is returned

Reply:
I will change it then to `string | null`. Thank you for the help!
It would be nice if we could see if we can find some time to upgrade to reCAPTCHA v3 😬 .